projects/orquestra-sdk/src/orquestra/sdk/_client/_base/_workflow.py

```diff
@@ -441,7 +441,12 @@ def model(self) -> ir.WorkflowDef:
                 "Parametrized WorkflowTemplates cannot be serialised yet"
             )
         else:
-            return self().model
+            # The reason we do type: ignore here is the following:
+            # self.__call__() accepts Paramspec _P as input types. But since we know
+            # that _P is empty (self.is_parametrized above doing as basically a type
+            # assertion) we can call __call__ that way, but  type checkers do not know
+            # that.
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** 
```suggestion
            # `WorkflowTemplate.__call__` accepts arguments if and only if the
            # underlying function has parameters.
            # The `self.is_parametrized` property tells us if the workflow function
            # is parametrized.
            # However, `self.__call__()` accepts a Paramspec `_P` as the input
            # parameter type (`_P.args` and `_P.kwargs`).
            # This causes a type error in Pyright because Pyright does not have the
            # information that `is_parametrized` tells us if `_P` has any members.
            # When `is_parametrized` is false, we know that `_P.args` and `_P.kwargs`
            # are empty.
            # This means, that when we are in this branch, we can be sure that
            # `__call__` does not accept any args or kwargs.
```